src/WebJobs.Script/Rpc/WebHostLanguageWorkerChannelManager.cs

```diff
@@ -55,9 +55,10 @@ private async Task<ILanguageWorkerChannel> InitializeLanguageWorkerChannel(strin
             try
             {
                 languageWorkerChannel = _languageWorkerChannelFactory.CreateLanguageWorkerChannel(scriptRootPath, runtime, null, 0);
+                // Keep track of language worker channel as soon as we might start it
+                // TODO: we should not make this something that we just need to do.
                 AddOrUpdateWorkerChannels(runtime, languageWorkerChannel);
                 await languageWorkerChannel.StartWorkerProcessAsync();
-                SetInitializedWorkerChannel(runtime, languageWorkerChannel);
```

Comment:
This is the biggest cause of hanging code - we allow things to go wrong in StartWorkerProcessAsync and the StartStream message, and if things go wrong, we never set the TaskCompletionSource for the language worker channel. This causes a potential deadlock in all code paths in this class that used to be "Async"